### PR TITLE
Add RX65N SPI (RSPI) driver support

### PR DESCRIPTION
**Pull request by Bhindhiya**
## Summary

Add SPI (Renesas SPI) driver support

## Impact

Porting the RX65N RSPI driver

### CODE REVIEW

File: arch/renesas/src/rx65n/rx65n_irq.c [hunk 1]
```diff
@@ -426,6 +426,74 @@ void up_disable_irq(int irq)
 
 #endif
 #endif
+
+#ifdef CONFIG_RX65N_RSPI0
+  if (irq == RX65N_SPRI0_IRQ)
+    {
+      ICU.IER[04].BIT.IEN6 = 0;
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_irq.c:433]: Avoid using number preceding of '0'! A number preceeding of 0 is an octal number, this way 08 and 09 are illegal.

File: arch/renesas/src/rx65n/rx65n_irq.c [hunk 2]
```diff
@@ -426,6 +426,74 @@ void up_disable_irq(int irq)
...
+    }
+
+  if (irq == RX65N_SPTI0_IRQ)
+    {
+      ICU.IER[04].BIT.IEN7 = 0;
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_irq.c:438]: ditto

File: arch/renesas/src/rx65n/rx65n_irq.c [hunk 3]
```diff
@@ -426,6 +426,74 @@ void up_disable_irq(int irq)
...
+  if (irq == RX65N_SPII0_IRQ)
+    {
+      ICU.GENAL0.BIT.EN16 = 0;
+    }
+#endif
+
+#ifdef CONFIG_RX65N_RSPI1
+  if (irq == RX65N_SPRI1_IRQ)
+    {
+      ICU.IER[05].BIT.IEN0 = 1;
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_irq.c:455]: ditto

File: arch/renesas/src/rx65n/rx65n_irq.c [hunk 4]
```diff
@@ -426,6 +426,74 @@ void up_disable_irq(int irq)
...
+    }
+
+  if (irq == RX65N_SPTI1_IRQ)
+    {
+      ICU.IER[05].BIT.IEN1 = 1;
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_irq.c:460]: ditto

File: arch/renesas/src/rx65n/rx65n_irq.c [hunk 5]
```diff
@@ -856,6 +924,74 @@ void up_enable_irq(int irq)
 
 #endif
 #endif
+
+#ifdef CONFIG_RX65N_RSPI0
+  if (irq == RX65N_SPRI0_IRQ)
+    {
+      ICU.IER[04].BIT.IEN6 = 1;
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_irq.c:931]: ditto

File: arch/renesas/src/rx65n/rx65n_irq.c [hunk 6]
```diff
@@ -856,6 +924,74 @@ void up_enable_irq(int irq)
...
+    }
+
+  if (irq == RX65N_SPTI0_IRQ)
+    {
+      ICU.IER[04].BIT.IEN7 = 1;
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_irq.c:936]: ditto

File: arch/renesas/src/rx65n/rx65n_irq.c [hunk 7]
```diff
@@ -856,6 +924,74 @@ void up_enable_irq(int irq)
...
+  if (irq == RX65N_SPII0_IRQ)
+    {
+      ICU.GENAL0.BIT.EN16 = 1;
+    }
+#endif
+
+#ifdef CONFIG_RX65N_RSPI1
+  if (irq == RX65N_SPRI1_IRQ)
+    {
+      ICU.IER[05].BIT.IEN0 = 1;
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_irq.c:953]: ditto

File: arch/renesas/src/rx65n/rx65n_irq.c [hunk 8]
```diff
@@ -856,6 +924,74 @@ void up_enable_irq(int irq)
...
+    }
+
+  if (irq == RX65N_SPTI1_IRQ)
+    {
+      ICU.IER[05].BIT.IEN1 = 1;
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_irq.c:958]: ditto

File: arch/renesas/src/rx65n/rx65n_rspi.c [hunk 9]
```diff
@@ -0,0 +1,2557 @@
...
+#ifdef CONFIG_RX65N_RSPI0
+  if (bus == RX65N_RSPI_CHANNEL0)
+    {
+      /* Configure Transmit empty buffer interrupt source bit */
+
+      IEN(RSPI0, SPTI0) = 0;  /* Disable SPTI0 interrupts */
+      IR(RSPI0, SPTI0)  = 0;  /* Clear interrupt flag */
+      IPR(RSPI0, SPTI0) = RX65N_RSPI_INTRRUPT_PRIO;
+      IEN(RSPI0, SPTI0) = 1;  /* Enable SPTI0 interrupt */
+      ICU.IER[04].BIT.IEN7 = 1;
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_rspi.c:2072]: ditto

File: arch/renesas/src/rx65n/rx65n_rspi.c [hunk 10]
```diff
@@ -0,0 +1,2557 @@
...
+
+      /* Configure Receive buffer full interrupt source bit */
+
+      IEN(RSPI0, SPRI0) = 0; /* Disable SPRI0 interrupts */
+      IR(RSPI0, SPRI0) = 0;  /* Clear interrupt flag */
+      IPR(RSPI0, SPRI0) = RX65N_RSPI_INTRRUPT_PRIO;
+      IEN(RSPI0, SPRI0) = 1; /* Enable SPRI0 interrupt */
+      ICU.IER[04].BIT.IEN6 = 1;
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_rspi.c:2080]: ditto

File: arch/renesas/src/rx65n/rx65n_rspi.c [hunk 11]
```diff
@@ -0,0 +1,2557 @@
...
+      IEN(RSPI1, SPTI1) = 1;  /* Enable SPTI1 interrupt */
+      ICU.IER[05].BIT.IEN1 = 1;
+
+      /* Configure Receive buffer full interrupt */
+
+      IEN(RSPI1, SPRI1) = 0;  /* Disable SPRI1 interrupts */
+      IR(RSPI1,  SPRI1) = 0;  /* Clear interrupt flag */
+      IPR(RSPI1, SPRI1) = RX65N_RSPI_INTRRUPT_PRIO;
+      IEN(RSPI1, SPRI1) = 1;  /* Enable SPRI1 interrupt */
+      ICU.IER[05].BIT.IEN0 = 1;
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_rspi.c:2109]: ditto

File: arch/renesas/src/rx65n/rx65n_rspi_sw.c [hunk 12]
```diff
@@ -0,0 +1,2154 @@
...
+#ifdef CONFIG_RX65N_RSPI0
+  if (bus == RX65N_RSPI_CHANNEL0)
+    {
+      /* Configure Transmit empty buffer interrupt source bit */
+
+      IEN(RSPI0, SPTI0) = 0;  /* Disable SPTI0 interrupts */
+      IR(RSPI0, SPTI0)  = 0;  /* Clear interrupt flag */
+      IPR(RSPI0, SPTI0) = RX65N_RSPI_INTRRUPT_PRIO;
+      IEN(RSPI0, SPTI0) = 1;  /* Enable SPTI0 interrupt */
+      ICU.IER[04].BIT.IEN7 = 1;
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_rspi_sw.c:1708]: ditto

File: arch/renesas/src/rx65n/rx65n_rspi_sw.c [hunk 13]
```diff
@@ -0,0 +1,2154 @@
...
+
+      /* Configure Receive buffer full interrupt source bit */
+
+      IEN(RSPI0, SPRI0) = 0; /* Disable SPRI0 interrupts */
+      IR(RSPI0, SPRI0) = 0;  /* Clear interrupt flag */
+      IPR(RSPI0, SPRI0) = RX65N_RSPI_INTRRUPT_PRIO;
+      IEN(RSPI0, SPRI0) = 1; /* Enable SPRI0 interrupt */
+      ICU.IER[04].BIT.IEN6 = 1;
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_rspi_sw.c:1716]: ditto

File: arch/renesas/src/rx65n/rx65n_rspi_sw.c [hunk 14]
```diff
@@ -0,0 +1,2154 @@
...
+#ifdef CONFIG_RX65N_RSPI1
+  if (bus == RX65N_RSPI_CHANNEL1)
+    {
+      /* Configure Transmit empty buffer interrupt */
+
+      IEN(RSPI1, SPTI1) = 0;  /* Disable SPTI1 interrupts */
+      IR(RSPI1, SPTI1)  = 0;  /* Clear interrupt flag */
+      IPR(RSPI1, SPTI1) = RX65N_RSPI_INTRRUPT_PRIO;
+      IEN(RSPI1, SPTI1) = 1;  /* Enable SPTI1 interrupt */
+      ICU.IER[05].BIT.IEN1 = 1;
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_rspi_sw.c:1737]: ditto

File: arch/renesas/src/rx65n/rx65n_rspi_sw.c [hunk 15]
```diff
@@ -0,0 +1,2154 @@
...
+
+      /* Configure Receive buffer full interrupt */
+
+      IEN(RSPI1, SPRI1) = 0;  /* Disable SPRI1 interrupts */
+      IR(RSPI1,  SPRI1) = 0;  /* Clear interrupt flag */
+      IPR(RSPI1, SPRI1) = RX65N_RSPI_INTRRUPT_PRIO;
+      IEN(RSPI1, SPRI1) = 1;  /* Enable SPRI1 interrupt */
+      ICU.IER[05].BIT.IEN0 = 1;
```
Comment (acassis) [inline arch/renesas/src/rx65n/rx65n_rspi_sw.c:1745]: ditto

**Review by acassis — CHANGES_REQUESTED** [review]

File: arch/renesas/src/rx65n/rx65n_rspi_sw.c (hunk 15 above)
Comment (Bhindhiya) [reply arch/renesas/src/rx65n/rx65n_rspi_sw.c]: Thank you for the comments.

The comments have been addressed in 285d16d.

**Pull request merged**